src/components/select-field/BaseSelectField.js

```diff
@@ -85,6 +85,24 @@ class BaseSelectField extends React.Component<Props, State> {
             isOpen: false,
             shouldScrollIntoView: false,
         };
+
+        this.fieldRef = React.createRef();
+    }
+
+    componentDidMount() {
+        this.calculateOffsetTopValue();
+    }
+
+    /**
+     * Calculate this field's distance from the top.
+     * This allows us to create an overflow container for the field's dropdown menu that is navigable using both
+     * the mouse and arrow keys and will always fit within the current viewport.
+     */
+    calculateOffsetTopValue() {
+        this.offsetTopValue =
```

Comment:
Minor, but I'm not sure this method is necessary. It seems like its logic could be inlined in `componentDidMount`. If we decide to keep it, I think it should just calculate and return the value rather than assigning it to the class instance, to reduce side effects and ease testing.